docs/content/docs/guides/migration_guide.md

```diff
@@ -0,0 +1,244 @@
+
+For instance, instead of:
+
+```python
+from outlines import from_vllm
+from vllm import LLM
+
+model = from_vllm(
+    LLM.from_pretrained("meta-llama/Llama-3.1-8B-Instruct")
+)
+model.load_lora("path/to/lora/file")
+
+response = model("foo")
+```
+
+You should now do:
+
+```python
+from outlines import from_vllm
+from vllm import LLM
+from vllm.lora.request import LoRARequest
+
+model = from_vllm(
+    LLM.from_pretrained("meta-llama/Llama-3.1-8B-Instruct")
+)
+lora_request = LoRARequest("path/to/lora/file", 1, "path/to/lora/file")
+
+response = model("foo", lora_request=lora_request)
+```
+
+The `ExLlamav2` model has been removed without replacement. This inference library is not fully compatible with Outlines, so we decided to remove it. You can still use it until final deprecation, but we recommend you to migrate to a different inference library right now.
+
+### Samplers
+
+The `outlines.samplers` module has been removed without replacement. You should now use the arguments of the inference library model to control the sampling. Depending on the model you use, this could be done at initialization or when calling the model to generate text (so when calling the outlines model or a generator).
+
+For instance, instead of:
+
+```python
+from outlines import generate
+
+model = <transformers_model>
+
+generator = generate.text(model, samplers.beam_search(2))
+response = generator("foo")
+```
+
+You should now do:
+
+```python
+from outlines import Generator
+
+model = <transformers_model>
+
+generator = Generator(model)
+response = generator("foo", num_beams=2)
+```
+
+### Functions
+
+The `outlines.function` module has been removed. It is replaced by the `outlines.applications` module. An `Application` serves a similar purpose as a `Function`: it encapsulates a prompt template and an output type. A difference is that can `Application` is not instantiated with a model name. Instead, you should provide a model instance along with the prompt when calling it.
+
+For instance, instead of:
+
+```python
+from outlines import Function
+
+prompt_template = ...
+output_type = ...
+
+fn = Function(
+    prompt_template,
+    output_type,
+    "hf-internal-testing/tiny-random-GPTJForCausalLM",
+)
+
+result = fn("foo")
+```
+
+You should now do:
+
+```python
+from outlines import Application
+
+prompt_template = ...
+output_type = ...
+
+application = Application(
+    prompt_template,
+    output_type,
+)
+
+model = ...
+
+result = application(model, "foo")
+```
+
+### Text generation return types
+
+In the previous version of Outlines, the return type of the generators depended on the output type provided. For instance, if you passed a Pydantic model to the `generate.json` function, the return type was a Pydantic model instance. In the v1, the return type of a generator is always a `str`, the raw text generated by the model. You are responsible for parsing the text into the desired format.
```

Comment:
Either we change this before releasing (cf Cameron's issue), or we can say "for now".